packages/nextjs/components/scaffold-eth/TokenAmount.tsx

```diff
@@ -0,0 +1,64 @@
+import { formatEther } from "viem";
+import { useAccountBalance } from "~~/hooks/scaffold-eth";
+import { useTargetNetwork } from "~~/hooks/scaffold-eth/useTargetNetwork";
+
+type TokenAmountProps = {
+  amount: number;
+  decimals?: number;
+  isEth?: boolean;
+};
+
+/**
+ * Display a token amount with a custom amount of decimals.
+ * If it's ETH, allow for showing its USD value with a button push.
+ */
+export const TokenAmount = ({ amount, decimals = 4, isEth = false }: TokenAmountProps) => {
+  const { targetNetwork } = useTargetNetwork();
+  const { price, isError, isLoading, onToggleBalance, isEthBalance } = useAccountBalance();
+
+  return (
+    <>
+      {isEth ? (
+        <>
+          {isLoading ? (
```

Comment:
`isLoading` and `isError` are about `balance` part of `useAccountBalance`, which you don't use. So as I understand you don't need these parts in the component

Reply:
++, I think we could remove `useAccountBalance` completely from here, we could get `price` from here :  `const price = useGlobalState(state => state.nativeCurrencyPrice);` 

The use `useAccountBalance` initially confused me alot while reviewing the code